src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] array;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        array = new Node[DEFAULT_CAPACITY];
+        threshold = 12;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = new Node<>(key, value, null);
+        if (size == threshold) {
+            resize();
+        }
+        int bucket = getKeyHash(key) % array.length;
+        Node currentNode = array[bucket];
+        if (array[bucket] == null) {
+            array[bucket] = node;
+        } else {
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, node.key)) {
+                    currentNode.value = node.value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            node.next = array[bucket];
+            array[bucket] = node;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int bucket = getKeyHash(key) % array.length;
+        Node<K, V> currentNode = array[bucket];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getKeyHash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() + 17);
```

Comment:
```suggestion
        return key == null ? 0 : Math.abs(key.hashCode() * 17);
```
Why + 17? Multiply or remove